### PR TITLE
fix/BannerTip to TS

### LINKED ISSUE #19119: Migrate components to TS: BannerTip
### Description
Migrate `BannerTip` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- Fixes #19119 

### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/e479481b-5b61-47a3-96cf-110f82b0f51b

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/71ca80b4-ed09-4b99-93eb-195b9f671b59

unit tests
image

## Manual Testing Steps

- Check storybook controls for `BannerTip`
- Ensure unit tests pass by running `yarn jest ui/components/component-library/banner-tip`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/banner-tip/banner-tip.stories.tsx [hunk 1]
```diff
@@ -1,32 +1,20 @@
...
-  6,
-  7,
-  8,
-  9,
-  10,
-  11,
-  12,
-  'auto',
-];
-
```
Comment (dhruvv173) [inline ui/components/component-library/banner-tip/banner-tip.stories.tsx:29]: Removed as this component extends `BannerBase` which extends the `Box` component and hence all the controls are available

File: ui/components/component-library/banner-tip/banner-tip.stories.tsx [hunk 2]
```diff
@@ -44,30 +32,15 @@ export default {
...
-    },
-    marginBottom: {
-      options: marginSizeControlOptions,
-      control: 'select',
-      table: { category: 'box props' },
-    },
-    marginLeft: {
-      options: marginSizeControlOptions,
-      control: 'select',
-      table: { category: 'box props' },
```
Comment (dhruvv173) [inline ui/components/component-library/banner-tip/banner-tip.stories.tsx:65]: After:
image

File: ui/components/component-library/banner-tip/banner-tip.stories.tsx [hunk 3]
```diff
@@ -147,7 +111,7 @@ ActionButton.args = {
     'Use actionButtonLabel for action text, actionButtonOnClick for the onClick handler, and actionButtonProps to pass any ButtonLink prop types such as iconName',
 };
```
Comment (dhruvv173) [inline ui/components/component-library/banner-tip/banner-tip.stories.tsx:112]: apart from this console error in story everything works fine. 
image

tried type assertion but it didnt work either. 
```
  actionButtonProps: {
    iconName: IconName.Arrow2Right,
    iconPositionRight: true,
  } as ButtonLinkProps<'button'>,
```

Comment (georgewrmarshall) [reply ui/components/component-library/banner-tip/banner-tip.stories.tsx:112]: Ah this looks like a mistake it should be `endIconName`

Comment (dhruvv173) [reply ui/components/component-library/banner-tip/banner-tip.stories.tsx:111]: done [`here`](https://github.com/MetaMask/metamask-extension/pull/20538/commits/574ec7f09f8c59f7d89da837f39ff9e1fe5a4611)

File: ui/components/component-library/banner-tip/README.mdx [hunk 4]
```diff
@@ -110,8 +110,7 @@ import { BannerTip, IconName } from '../../component-library';
   title="Action prop demo"
   actionButtonLabel="Action"
   actionButtonProps={{
-    icon: IconName.Arrow2Right, // TODO: change to iconName
-    iconPositionRight: true,
+    endIconName: IconName.Arrow2Right,
```
Comment (dhruvv173) [inline ui/components/component-library/banner-tip/README.mdx:113]: After:
image

File: ui/components/component-library/banner-tip/banner-tip.test.tsx [hunk 5]
```diff
@@ -39,7 +45,9 @@ describe('BannerTip', () => {
     expect(getByTestId('banner-tip-greeting')).toHaveClass(
       'mm-banner-tip--logo',
     );
+    expect(getByTestId('banner-tip-greeting')).toHaveClass('custom-logo-class');
     expect(getByTestId('banner-tip-chat')).toHaveClass('mm-banner-tip--logo');
+    expect(getByTestId('banner-tip-chat')).toHaveClass('custom-logo-class');
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner-tip/banner-tip.test.tsx:50]: Updating test to ensure the logo className isn't overwritten when a custom class is added

**Review by georgewrmarshall — APPROVED** [review]
LGTM! I pushed some small updates that I think were already existing issues not related to the migration. Thanks for your contribution @dhruvv173!

**Review by garrettbear — APPROVED** [review]

**Pull request merged**